pkg/apis/garden/validation/validation.go

```diff
@@ -295,7 +303,34 @@ func validateKubernetesConstraints(kubernetes garden.KubernetesConstraints, fldP
 	return allErrs
 }
 
-func validateMachineTypeConstraints(machineTypes []garden.MachineType, fldPath *field.Path) field.ErrorList {
+func validateKubernetesSettings(kubernetes garden.KubernetesSettings, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(kubernetes.Versions) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("versions"), "must provide at least one Kubernetes version"))
+	}
+
+	latestKubernetesVersion, err := helper.DetermineLatestExpireableVersion(kubernetes.Versions)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, latestKubernetesVersion, "failed to determine latest kubernetes version from cloud profile"))
+	}
+
+	if latestKubernetesVersion.ExpirationDate != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("versions[]").Child("expirationDate"), latestKubernetesVersion.ExpirationDate, fmt.Sprintf("expiration date of latest kubernetes version ('%s') must not be set", latestKubernetesVersion.Version)))
+	}
+
+	r, _ := regexp.Compile(`^([0-9]+\.){2}[0-9]+$`)
```

Comment:
Can't we use semver for the kubernetes version also instead of matching regex?

Reply:
No, the patch version is defaulted to `0` if you use `1.11` for example. That should not be allowed.